url8012c8d031_dmbb966_BBMS_tgz-pJ8-testbed_JNEATtestJ8/src/gui/GUIMenu.java

```diff
@@ -0,0 +1,1199 @@
+			GUI_NB.GCO("Clicking will now set terrain to TREES");
+			GlobalFuncs.placeUnit = 11;
+		}
+	}
+	
+	/** Sets the mode so that clicking will change the given terrain to TALL GRASS
+	 */
+	public static class ModePaintTallGrass implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			GUI_NB.GCO("Clicking will now set terrain to TALL GRASS");
+			GlobalFuncs.placeUnit = 12;
+		}
+	}
+	
+	/** Sets the mode so that clicking will change the given hex to a vapor source
+	 */
+	public static class ModeSetVaporSource implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			GUI_NB.GCO("Clicking will now set hex to VAPOR SOURCE");
+			GlobalFuncs.placeUnit = 21;
+			new DisplayModeVapor().actionPerformed(event);
+		}
+	}
+	
+	/** Sets the mode so that clicking will change the given hex to a vapor sink
+	 */
+	public static class ModeSetVaporSink implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			GUI_NB.GCO("Clicking will now set hex to VAPOR SINK");
+			GlobalFuncs.placeUnit = 22;
+			new DisplayModeVapor().actionPerformed(event);
+		}
+	}
+	
+	/** Sets the mode so that clicking will remove a vapor source or sink
+	 */
+	public static class ModeSetVaporNorm implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			GUI_NB.GCO("Clicking will now REMOVE vapor source or sink");
+			GlobalFuncs.placeUnit = 20;
+			new DisplayModeVapor().actionPerformed(event);
+		}
+	}
+	
+	
+	
+	/** Outputs all spotting this turn to the GUI console
+	 */
+	public static class DisplaySpotsThisTurn implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			spotting.SpotRecords x = GlobalFuncs.allSpots.getReportsTime(Clock.time);
+			GUI_NB.GCO(x.DisplayRecords());
+		}
+	}
+	
+	/** Outputs all spotting data to the GUI console
+	 */
+	public static class DisplayAllSpots implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			spotting.SpotRecords x = GlobalFuncs.allSpots.getReportsTime(0, Clock.time);
+			GUI_NB.GCO(x.DisplayRecords());
+		}
+	}
+	
+	/** Outputs all spotting data for the selected unit to the GUI console
+	 */
+	public static class DisplayAllSpotsForUnit implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			if (GlobalFuncs.selectedUnit != null) {
+				spotting.SpotRecords x = GlobalFuncs.allSpots.getReportsTarget(GlobalFuncs.selectedUnit);
+				GUI_NB.GCO(x.DisplayRecords());
+			}
+			else {
+				GUI_NB.GCO("ERROR: No unit selected!");
+			}
+			
+		}
+	}
+	
+	/** Sets the display mode to UNIT in the main display and info panes
+	 */
+	public static class DisplayModeUnit implements ActionListener {
+		public void actionPerformed(ActionEvent event) {
+			GUI_NB.GCO("Display Mode set to UNIT");
+			GlobalFuncs.showVapor = false;
+			GlobalFuncs.showLOS = true;
+			GlobalFuncs.showFOW = true;
+			GUIInfoPane.changePaneMode(DetailedInfoEnum.UNIT);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 265:269, 277:281, 287:291.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.